skjoedt/homelab-test
# Add initial cluster configuration

This PR adds the initial infrastructure configuration for the k3s cluster including:

- kube-vip configuration for HA control plane (VIP: 192.168.68.20)
  - Configured to use enp5s0 interface
- Traefik configuration with LoadBalancer IP 10.0.10.2
- Test website deployment accessible at test.fanen.home

Key configurations:
- Using interface enp5s0 for kube-vip
- Traefik deployed as DaemonSet
- Test website with 2 replicas

Prerequisites for deployment:
1. Configure local dnsmasq to resolve test.fanen.home to 10.0.10.2
2. Ensure k3s nodes network interface enp5s0 is correctly configured
3. Have helmfile installed locally

Testing:
1. After applying, verify kube-vip is working by checking if the VIP is accessible
2. Verify Traefik gets the correct LoadBalancer IP
3. Test website should be accessible at http://test.fanen.home
